### ISSUE TITLE
Not generating types for a Materialized View

### ISSUE THREAD

**Issue by dargue3** [RobinBlomberg/kysely-codegen#72]
Hey! Thanks for the repo it's been saving me so much time!

I've got a table and a corresponding materialized view that I've spun off of it to add further indices. It seems like the codegen is not picking up on this new view and I was wondering if that's a limitation or just something I'm missing in the options? Similar to the following:

```sql
CREATE TABLE users
(
    id   INT  NOT NULL,
    name TEXT NOT NULL
);

CREATE MATERIALIZED VIEW users_mv AS
SELECT id,
       name
FROM users;
```

My `kysely-codegen` command is like this:
```sh
./node_modules/.bin/kysely-codegen \
  --camel-case \
  --dialect postgres \
  --url $DATABASE_URL \
  --out-file $SCHEMA_PATH
```

Once I've done this, I correctly see the types for `Users` however there are no types for the materialize view, and thus, when I do something like `db.selectFrom('users_mv')` it's giving me a TypeError.

**Comment by virtuallyunknown** [comment]
I would also be interested in this feature.

**Comment by virtuallyunknown** [comment]
I am leaving this comment as a temporary workaround if someone needs this, until hopefully materialized views get implemented in a future version.

1. In your postgres database, create a normal view.
2. Create exactly the same view but materialized.
3. Extend the types generated by `kysely-codegen` before passing them to your `kysely` instance (thanks @igalklebanov for the idea).

```SQL
CREATE OR REPLACE VIEW my_view AS ...;
CREATE MATERIALIZED VIEW IF NOT EXISTS my_materialized_view AS ...;
```

```ts
import pg from 'pg';
import { CamelCasePlugin, Kysely, PostgresDialect, sql } from 'kysely';
import type { DB } from '../types/index.js';
//            ^^ types generated by kysely-codegen

const { Pool } = pg;

interface DBWithMaterializedViews extends Omit<DB, 'myView'> {
    myMaterializedView: DB['myView']
    // or you can make all props non-nullable
    myMaterializedView: { [K in keyof DB['myView']]: NonNullable<DB['myView'][K]> }
}

export const db = new Kysely<DBWithMaterializedViews>({
    dialect: new PostgresDialect({
        pool: new Pool({
            // credentials
        }),
    }),
    plugins: [new CamelCasePlugin()],
    log: ['query'],
});
```

---

In short, we're creating the `my_view` type solely for the purpose of having `kysely-codegen` generate types for it, then extending the generated types (`DB`) with `myMaterializedView` (which is the name of the materialized view in camelCase) before passing this to the kysely instance.

**Comment by will-weiss** [comment]
I'd also love this, but at least for postgres we may be at the mercy of the database itself. [This StackOverflow post](https://dba.stackexchange.com/questions/285302/why-columns-in-views-are-always-nullable) is relevant, as it explains that columns in views are (always?) nullable.

**Comment by virtuallyunknown** [comment]
> I'd also love this, but at least for postgres we may be at the mercy of the database itself. [This StackOverflow post](https://dba.stackexchange.com/questions/285302/why-columns-in-views-are-always-nullable) is relevant, as it explains that columns in views are (always?) nullable.

This is correct, but having nullable types is still way better than having no types at all. Kanel-kysely for example generates materialized views like this:

```ts
export default interface SomeMV {
  foo: ColumnType<number | null, never, never>;
  bar: ColumnType<string | null, never, never>;
  baz: ColumnType<DBPlatformName | null, never, never>;
}
```

**Comment by gvko** [comment]
I'd very much like this feature as well! 🙏

**Comment by ladrone-vincet** [comment]
Any new updates or best practices about this?

**Comment by VastBlast** [comment]
Since this library does not appear to be actively maintained, I started a fork (for now) that adds support for materialized views, along with other features and bug fixes based on issues in this repo. Feel free to open issues there if you run into anything, at least for the time being.

I’m mainly using this in my own projects, but wanted to share in case others find it useful:

[kysely-generate](https://github.com/VastBlast/kysely-generate)
